src/main/java/TaskList.java

```diff
@@ -0,0 +1,135 @@
+    return "Got it. I've added this task:\n" +  tasks.get(tasks.size() - 1) + "\nNow you have "
+            + tasks.size() + " tasks in the list.";
+  }
+
+  /**
+   * Handles the addition of a deadline
+   *
+   * @param inputs Userinput split into tokens
+   */
+  public String handleDeadline(String[] inputs) {
+    String[] tokens = inputs[1].split(" /by ");
+    tasks.add(new Deadline(tokens[0], tokens[1]));
+    System.out.println("Got it. I've added this task:");
+    System.out.println(tasks.get(tasks.size() - 1));
+    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    return "Got it. I've added this task:\n" +  tasks.get(tasks.size() - 1) + "\nNow you have "
+            + tasks.size() + " tasks in the list.";
+  }
+
+  /**
+   * Handles the addition of a todo
+   *
+   * @param inputs Userinput split into tokens
+   */
+  public String handleToDo(String[] inputs) throws DukeException {
+    if (inputs.length < 2) {
+      throw new DukeException("The description of a todo cannot be empty.");
+    }
+    tasks.add(new ToDo(inputs[1]));
+    System.out.println("Got it. I've added this task:");
+    System.out.println(tasks.get(tasks.size() - 1));
+    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    return "Got it. I've added this task:\n" +  tasks.get(tasks.size() - 1) + "\nNow you have "
+            + tasks.size() + " tasks in the list.";
+  }
+
+  /**
+   * Marks a given task as Done
+   *
+   * @param inputs Userinput split into tokens
+   */
+  public String handleDone(String[] inputs) throws DukeException {
+    int index = Integer.parseInt(inputs[1]) - 1;
+    if (index >= tasks.size() || index < 0) {
+      throw new DukeException("This task does not exist");
+    }
+    tasks.get(index).markDone();
+    System.out.println("Nice! I've marked this task as done: ");
+    System.out.println(tasks.get(index));
+    return "Nice! I've marked this task as done: \n" + tasks.get(index);
+  }
+
+  /**
+   * Deletes a specified task
+   *
+   * @param inputs Userinput split into tokens
+   */
+  public String handleDelete(String[] inputs) throws DukeException {
+    int index = Integer.parseInt(inputs[1]) - 1;
+    if (index >= tasks.size() || index < 0) {
+      throw new DukeException("This task does not exist");
+    }
+    System.out.println("Noted. I've removed this task: ");
+    System.out.println(tasks.get(index));
+    String oldIndex = tasks.get(index).toString();
+    tasks.remove(index);
+    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    return "Noted. I've removed this task: \n" + oldIndex + "\nNow you have "
+            + tasks.size() + " tasks in the list.";
+  }
+
+  /** Prints out the TaskList */
+  public String list() {
+    StringBuilder out = new StringBuilder("Task list\n");
+    System.out.println("Task list");
+    int i = 0;
+    for (Task task : tasks) {
+      System.out.println(++i+ ". " + task);
```

Comment:
Should there be a space after `i` such that `System.out.println(++i + ". " + task);`

Reply:
yep, thats a good point